### DISCUSSION TITLE
Updating deprecated set-env in release workflow

### DISCUSSION THREAD

**Discussion by glennzw** [community/community#32544] [Code Security] [answered]
Hi all,

My Workflow release builds are failing with the error:

```
Error: Unable to process command '::set-env name=RELEASE::gophish-v0.12.0-linux-64bit' successfully.
Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
```

via this Workflow script:

https://github.com/gophish/gophish/actions/runs/2852268619/workflow#L45

Following the [documentation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) link from the above [security release](https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/) environmental variables should now be set like so:

```
echo "{environment_variable_name}={value}" >> $GITHUB_ENV
```

My question is; would the below be a correct fix for my [release.yml](https://github.com/gophish/gophish/actions/runs/2852268619/workflow#L45)

From this:
```
run: echo "::set-env name=RELEASE::gophish-${{ github.event.release.tag_name }}-${{ matrix.releaseos}}-32bit"
```

to this:

```
echo "{RELEASE}={gophish-${{ github.event.release.tag_name }}-${{ matrix.releaseos}}-32bit}" >> $GITHUB_ENV

```

As a follow up question; is there a way to test my release workflow beyond committing a few file? It'd be a shame to have to make multiple commits for typos/etc.

Thank you!
G

**Comment by airtower-luna** [comment, marked as answer]
No, the `{}` in the documentation quote is just an odd way to mark placeholders. So you need something like this:

```sh
echo "RELEASE=gophish-${{ github.event.release.tag_name }}-${{ matrix.releaseos }}-32bit" >> $GITHUB_ENV
```

> As a follow up question; is there a way to test my release workflow beyond committing a few file? It'd be a shame to have to make multiple commits for typos/etc.

I'd add a different trigger (e.g. `push` or `workflow_dispatch`) and push to a branch to test. Naturally you won't have the `github.event.release.tag_name` then, but you can test the same pattern with some other variable.

**Comment by glennzw** [comment]
Great thanks so much @airtower-luna, that worked! I'm stuck on a new problem after this update, if you'd have time to give me a hand: https://github.com/orgs/community/discussions/32786
